packages/nestjs/src/setup.ts

```diff
@@ -123,6 +124,36 @@ class SentryGlobalGraphQLFilter {
 Catch()(SentryGlobalGraphQLFilter);
 export { SentryGlobalGraphQLFilter };
 
+/**
+ * Global filter to handle exceptions and report them to Sentry.
+ *
+ * This filter is a generic filter that can handle both HTTP and GraphQL exceptions.
+ */
+class SentryGlobalGenericFilter {
+  public readonly __SENTRY_INTERNAL__: boolean;
+  private readonly _httpFilter: SentryGlobalFilter;
+  private readonly _graphqlFilter: SentryGlobalGraphQLFilter;
+
+  public constructor(applicationRef?: HttpServer) {
+    this.__SENTRY_INTERNAL__ = true;
+    this._httpFilter = new SentryGlobalFilter(applicationRef);
+    this._graphqlFilter = new SentryGlobalGraphQLFilter();
+  }
+
+  /**
+   * Catches exceptions and reports them to Sentry unless they are HttpExceptions.
+   */
+  public catch(exception: unknown, host: ArgumentsHost): void {
+    if (host.getType<'graphql'>() === 'graphql') {
```

Comment:
We could think about checking for both host types 'graphql' and 'http' and otherwise provide a warning for an unsupported host type, wdyt?

Reply:
I like the idea, but the `SentryGlobalFilter` can be used for other host types as well (like rpc), since it extends the `BaseExceptionFilter` from nestjs. Maybe a rename from `_httpFilter` to `_globalFilter` would be more suitable.

Reply:
I changed the filter to extend the base filter and only use the graphql filter when the host type is `graphql`